File: cpp/sketch/enum.cpp
#include <iostream> /* C++ 标准输入输出头文件 */

/**************************************************************************************************/
enum class WeekDay {
    Sunday,
    Monday,
    Tuesday,
    Wednesday,
    Thursday,
    Friday,
    Saturday
};

static const char* weekday_name(WeekDay day) {
    switch (day) {
        case WeekDay::Sunday: return "Sunday"; break;
        case WeekDay::Monday: return "Monday"; break;
        case WeekDay::Tuesday: return "Tuesday"; break;
        case WeekDay::Wednesday: return "Wednesday"; break;
        case WeekDay::Thursday: return "Thursday"; break;
        case WeekDay::Friday: return "Friday"; break;
        case WeekDay::Saturday: return "Saturday"; break;
        default: return "[Error]"; break;
    }
}

/**************************************************************************************************/
int main(int argc, char* argv[]) {
    WeekDay day = WeekDay::Tuesday;

    if (argc > 1) {
        day = static_cast<WeekDay>(std::stoi(argv[1]));
    }
    
    std::cout << "今天是 " << weekday_name(day) << "，";
    std::cout << "它的整型值是" << static_cast<int>(day) << std::endl;
    
    return 0;
}

File: py/sketch/enum.py
#!/usr/bin/env python3

import sys  # 系统相关参数和函数
import enum # 枚举模块

################################################################################
class WeekDay(enum.Enum):
    Sunday = 0
    Monday = 1
    Tuesday = 2
    Wednesday = 3
    Thursday = 4
    Friday = 5
    Saturday = 6

################################################################################
def main(argc, argv):
    day = WeekDay.Tuesday

    if argc > 1:
        day = WeekDay(int(argv[1]))

    print("今天是 {}({}), 它的整型值是 {}。".format(day.name, type(day), day.value))

    return 0



################################################################################
if __name__ == "__main__":
    sys.exit(main(len(sys.argv), sys.argv))
